File: frontend/src/types/Zone.ts
import { Point } from './Point';

export interface Zone {
    id: number;
    region: Point[];
    type: ZoneType;
}

export enum ZoneType {
    None = "None",
    Restricted = "Restricted",
    Urban = "Urban",
    Available = "Available",
}

export const MAIN_ZONE_TYPE = ZoneType.None;

File: backend/PathScape.WebApi/Dto/Zone.cs
using System.Collections.Generic;
using System.Text.Json.Serialization;
using PathScape.Domain.Models;

namespace WebApplication2.Dto;

public class Zone
{
    [JsonPropertyName("id")]
    public int Id { get; set; }
    
    [JsonPropertyName("region")]
    public required IEnumerable<Point> Region { get; init; }
    
    [JsonPropertyName("type")]
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public required ZoneType ZoneType { get; init; }
}